.github/workflows/cypress.yml

```diff
@@ -0,0 +1,42 @@
+name: Run Cypress Tests
+
+on:
+  push:
+    branches:
+      - main 
+  pull_request: 
+
+jobs:
+  test:
+    runs-on: ubuntu-latest
+
+    steps:
+    # Checkout the code
+    - name: Checkout code
+      uses: actions/checkout@v3
+
+    # Set up Node.js
+    - name: Setup Node.js
+      uses: actions/setup-node@v3
+      with:
+        node-version: 18 
+
+    # Install dependencies
+    - name: Install dependencies
+      run: npm install
+
+    # Install Cypress
+    - name: Install Cypress
+      run: npm install cypress --save-dev
+
+    # Run Cypress tests
+    - name: Run Cypress Tests
+      run: npx cypress run
```

Comment:
I used npx cypress run here for simplicity and to directly call Cypress without relying on the package.json scripts. However, It was an oversight since I wrote custom scripts in package.json, I could have used them instead for better consistency and to leverage the project’s configuration. I’ll update the workflow to use the existing scripts.